kyuubi-server/src/main/scala/org/apache/kyuubi/credentials/HadoopCredentialsManager.scala

```diff
@@ -181,12 +182,14 @@ class HadoopCredentialsManager private (name: String) extends AbstractService(na
   }
 
   // Visible for testing.
-  private[credentials] def getOrCreateUserCredentialsRef(appUser: String): CredentialsRef =
+  private[credentials] def getOrCreateUserCredentialsRef(
+      appUser: String,
+      waitComplete: Boolean = false): CredentialsRef =
     userCredentialsRefMap.computeIfAbsent(
```

Comment:
we need check whether the existing `CredentialsRef` has valid `encodedCredentials`, it might be null.